### DISCUSSION TITLE
How to hide the first header row of the column groups?

### DISCUSSION THREAD

**Discussion by yaronlevi** [TanStack/table#2096] [General] [answered]
Hi (-:

I hope you are feeling good these days.

In a simple table like this one:

<img width="637" alt="Screen Shot 2020-04-03 at 13 53 25" src="https://user-images.githubusercontent.com/5398977/78353436-b46ea500-75b2-11ea-9429-77e4244a8d66.png">

How can we hide the entire first row that contains the group names? (the row that contains "Name" and "Info").

Thanks for any help on the issue.

**Comment by tannerlinsley** [comment, marked as answer]
Unnest the columns from the parent column groups. So in other words, flatten the columns definition.

**Reply by gschauhan** [reply]
This does not work with typescript. I get incompatible type error. I am trying to use basic example from [quick start](https://react-table.tanstack.com/docs/quick-start). 
`Type '{ columns: { Header: string; accessor: string; }[]; }[]' is not assignable to type 'Column<{ col1: string; col2: string; }>[]'.`

If I use 
`   const columns = React.useMemo(
     () => [
       {
         Header: 'Column 1',
         accessor: 'col1', // accessor is the "key" in the data
       },
       {
         Header: 'Column 2',
         accessor: 'col2',
       },
     ],
     []
   )`
    I get above specified error, so I have to add something like 
    
    `const columns = React.useMemo(
    () => [
      {
        Header: 'Name',
        columns: [
          {
            Header: 'First Name',
            accessor: 'col1',
          },
          {
            Header: 'Last Name',
            accessor: 'col2',
          },
        ],
      },
    ],
    [],
  );`
  
  so, How to hide the first header row of the column groups in typescript?

**Reply by philsp** [reply]
same problem here

**Reply by ghost** [reply]
having the same typescript error.

**Reply by alienriquebm** [reply]
same problem

**Reply by tannerlinsley** [reply]
Despite the community types for v6 being wrong, unnesting is still the proper way to do this. V8 is the same answer as well, but with the types being correct.

**Reply by alienriquebm** [reply]
Thanks man. How can avoid the TS error?

**Comment by yaronlevi** [comment]
Great thank you!

**Comment by taliribeiro** [comment]
I extend a Column interface with ColumnInterface from react-table and than worked for me.

```
export interface Column extends ColumnInterface {
  Header: string;
  accessor: string;
}
```
